packages/toolpad-studio/src/toolpad/propertyControls/SelectOptions.tsx

```diff
@@ -47,17 +48,35 @@ function SelectOptionsPropEditor({
     return null;
   }, [editingIndex, value]);
 
+  const validateOptionValue = React.useCallback(
+    (inputValue: string) => {
+      const errorState = value.some((item) =>
+        typeof item !== 'string' ? item.value === inputValue : item === inputValue,
+      );
+      if (errorState) {
+        setOptionErrorMessage('Must not have duplicate values');
+      } else {
+        setOptionErrorMessage('');
+      }
+    },
+    [value],
+  );
+
   const handleOptionTextInput = React.useCallback(
     (event: React.KeyboardEvent<HTMLInputElement>) => {
+      validateOptionValue((event.target as HTMLInputElement).value);
```

Comment:
Does it make sense to only validate the option when the user attempts to add, versus running the validation on every keystroke?